packages/rs-sdk/src/sync.rs

```diff
@@ -187,21 +190,91 @@ where
         async move {
             let settings = closure_settings.load_full().clone();
             let mut func = inner_fn.lock().await;
-            (*func)(*settings).await
+            let result = (*func)(*settings).await;
+
+            // Ban or unban the address based on the result
+            match &result {
```

Comment:
Maybe this logic should be converted into a function shared between dapi-client and sdk, to not repeat similar code? not sure about that, just an idea